### ISSUE TITLE
Spotted inaccurate reaching

### ISSUE THREAD

**Issue by pattacini** [robotology/icub-gazebo-grasping-sandbox#45]
### Bug description

While releasing the latest version, I've spotted that reaching is not as accurate as it was in the past.
Thus, further investigation should be required.

### Steps to reproduce

Just run the demo.

**Comment by pattacini** [comment]
FYI
@xEnVrE

**Comment by xEnVrE** [comment]
Hi @pattacini, thanks for letting me know.

Is this happening after the update to the latest distro release?

**Comment by pattacini** [comment]
> Is this happening after the update to the latest distro release?

Not sure about that but, given that the general idea behind the sandbox is reproducibility, I can easily roll it back to a previous distro. I'd only take some time to craft a systematic test.

**Comment by pattacini** [comment]
By means of the test infrastructure I added via #46, I managed to perform some verifications with different distros.

| Distro[^1] | Pre-grasp pose error[^2] | Grasp pose error[^2] |
| :---: | :--- | :--- |
| `v2022.09.1` | (0.0087119, 1.56492) | (0.00315036, 0.413285) |
| `v2022.11.0` | (0.00896821, 1.51956) | (0.00316361, 0.435019) |
| `v2023.02.0` | **(0.0348022, 4.04909)** | **(0.00304989, 2.53692)** |
| `v2023.02.3` | (0.0358665, 4.12649) | (0.00316696, 2.6221) |
| `v2023.05.0` | (0.0356232, 4.11189) | (0.00311436, 2.60816) |

[logs.zip](https://github.com/robotology/icub-gazebo-grasping-sandbox/files/11866086/logs.zip)

Clearly, something happened between `v2022.11.0` and `v2023.02.0`.
I'll keep investigating these differences for `YARP`, `ICUB`, `icub-models`, `yarp-gazebo-plugins`...

[^1]: Dependencies installed via Docker remain unvaried (e.g., Gazebo), so as the sandbox. Only the packages installed via the `robotology-superbuild` change according to their versions.
[^2]: Errors are given as pairs (position [m], orientation [deg]).

**Comment by pattacini** [comment]
For the ICUB package: https://github.com/robotology/icub-main/compare/v2.0.2...v2.1.0.
We don't have relevant changes (we just got rid of `iCub_SIM`).

**Comment by pattacini** [comment]
For GazeboYARPPlugins package: https://github.com/robotology/gazebo-yarp-plugins/compare/v4.5.2...v4.6.0.
Changes are regarded with `HandMk5CouplingHandler` only.

**Comment by pattacini** [comment]
For `icub-models` package: https://github.com/robotology/icub-models/compare/v1.26.0...v2.0.0.
Changes look massive. I'd need to inquire about them deeper.

**Comment by pattacini** [comment]
> For the `icub-models` package: [robotology/icub-models@`v1.26.0...v2.0.0`](https://github.com/robotology/icub-models/compare/v1.26.0...v2.0.0). Changes look massive. I'd need to inquire about them deeper ❓

By sticking to v2023.02.0 and downgrading only `icub-models` to `v1.26.0`, I got the following errors:

```console
[ERROR] |yarp.dev.PolyDriver|controlboardwrapper2| Could not find device <controlboardwrapper2>
[ERROR] |gazebo-yarp-plugins.plugins.GazeboYarpControlBoard| wrapper did not open, load failed.
```

So, [robotology/icub-models@`v1.26.0...v2.0.0`](https://github.com/robotology/icub-models/compare/v1.26.0...v2.0.0) involved prominently the use of the new NWS/NWC layers, whereas the deprecated device `controlboardwrapper2` was removed right in `yarp@3.8.0`.

Also, I spotted this strange change in the model used by the sandbox:
[...zeboV2_5_visuomanip/gazebo_icub_torso.ini → ...azeboV2_5_visuomanip/gazebo_icub_eyes.ini](https://github.com/robotology/icub-models/compare/v1.26.0...v2.0.0#diff-8828369fbfda07ba73a94f43d4d73434a432067e49e2202f8c98104d83739ee6)

**Comment by traversaro** [comment]
Perhaps the related PR is https://github.com/robotology/icub-models/pull/188 ?

**Comment by pattacini** [comment]
My hypothesis is that it was decided in `v2.0.0` to use the [conf params of the "official model" for the torso][v2], instead of sticking to the ["manual" version][v1] available in `v1.26.0`.

I'm going to verify it with a test.

[v2]: https://github.com/robotology/icub-models/blob/v2.0.0/iCub/conf/gazebo_icub_torso.ini#L15-L26
[v1]: https://github.com/robotology/icub-models/blob/v1.26.0/iCub_manual/conf_manual/iCubGazeboV2_5_visuomanip/gazebo_icub_torso.ini#L37-L48

**Comment by pattacini** [comment]
> I'm going to verify it with a test.

I can confirm that `v2023.05.0` using these [PID gains](https://github.com/robotology/icub-models/blob/v1.26.0/iCub_manual/conf_manual/iCubGazeboV2_5_visuomanip/gazebo_icub_torso.ini#L37-L48) works nicely.

I'll open up a PR for this.
cc @traversaro

**Comment by pattacini** [comment]
> I'll open up a PR for this.

Here it is:
- https://github.com/robotology/icub-models-generator/pull/241.

**Comment by pattacini** [comment]
Running the sandbox again in its normal configuration confirmed that the reaching is now as accurate as it was before.

**Comment by pattacini** [comment]
>To have the latest docker up-to-date, we need to wait until v2023.05.1 is available.

See https://github.com/robotology/icub-models-generator/pull/241#issuecomment-1628623554.

**Comment by pattacini** [comment]
> > To have the latest docker up-to-date, we need to wait until v2023.05.1 is available.
> 
> See [robotology/icub-models-generator#241 (comment)](https://github.com/robotology/icub-models-generator/pull/241#issuecomment-1628623554).

This issue should be solved via 2023.05.2.
